Makefile-eh

```diff
@@ -667,6 +665,12 @@ include_dirs:
 	# Remove selected header files.
 	$(RM) $(patsubst %,$(SYSROOT_INC)/%,$(MUSL_OMIT_HEADERS))
 
+	find $(SYSROOT_INC) -wholename '*.h' -exec sed -i '1i\
```

Comment:
I'm strictly against this change, since if no headers are included, then conditional compilation will break. This can be the case if, for example, someone does:

```C
#ifdef __wasix__
#include <somethingsomething.h>
#endif
```

at the start of a file. For now, `__wasi__` is quite sufficient for our usecases, and `__wasix__` should only be added at the compiler level once we start implementing a proper wasix target in clang.

Reply:
As I said earlier, until we have a proper target in Clang, this is the only reliable way to do this. Once we have support in Clang, I will remove this here.

Our `libffi` port depends on this being defined. Getting our changes merged into libffi will be a lot easier if we can demonstrate that they work without any compiler wrappers or extra defines.

Can we please merge this in for now? I'll open an issue for removing it again.

Reply:
I removed the `__wasix__` and will use `__has_include` to check for the presence of wasix